### PR TITLE
fix(core): reject flowId filter without namespace for executions

### PR DESCRIPTION
**Pull request by rishi-jat**
###  Description
What does this PR change?  
While investigating this issue, I found that there are two paths that currently allow querying executions by flowId without a namespace.

This PR addresses those two paths by:

- Rejecting FLOW_ID without NAMESPACE during QueryFilter validation for Resource.EXECUTION.
- Adding a defensive check in the legacy JDBC query path so flowId cannot be used without namespace.
- Updating the query construction so the flow_id predicate is only applied together with the namespace predicate.
- Adding tests covering the new validation and the legacy repository path.

I intentionally did not include the index merge or any SQL migration in this PR. Based on the discussion in the issue, the actual index change is still blocked by a product decision. This PR only implements the validation and query changes needed to enforce the expected behavior before any index changes are considered.

###  Related Issue
Pat-of #17074 

### Backend Checklist
- [x] Code compiles successfully and passes all checks
- [x] All unit and integration tests pass

### Additional Notes
From my investigation, I couldn’t find any OSS callers that intentionally use flowId without namespace. The changes are limited to enforcing this constraint consistently in both the QueryFilter-based API and the legacy repository path without introducing the index changes discussed in the issue.

### CODE REVIEW

File: core/src/main/java/io/kestra/core/models/QueryFilter.java [hunk 1]
```diff
@@ -844,4 +845,24 @@ private static void collectValidationErrors(QueryFilter filter, Resource resourc
...
+     */
+    private static void collectCrossFieldErrors(List<QueryFilter> filters, Resource resource, List<String> errors) {
+        if (Resource.EXECUTION == resource && hasField(filters, Field.FLOW_ID) && !hasField(filters, Field.NAMESPACE)) {
+            errors.add(
+                "Field FLOW_ID requires NAMESPACE to be specified for resource EXECUTION"
+            );
+        }
+    }
+
 }
```
[the remark this answers is not in the document]
Comment (rishi-jat) [reply core/src/main/java/io/kestra/core/models/QueryFilter.java:888]: This is already documented in the method's Javadoc as a known limitation. The flat existence check covers all realistic query patterns — no caller constructs (namespace = X) OR (flowId = Y). Full structural tree validation would add significant complexity for an edge case that doesn't occur in practice. Happy to revisit if it becomes a real concern.

File: core/src/test/java/io/kestra/core/models/QueryFilterTest.java [hunk 2]
```diff
@@ -821,6 +822,88 @@ void shouldNotThrowExceptionWhenRegexIsSafe() {
...
+        // Given a FLOW_ID filter without NAMESPACE for a non-EXECUTION resource
+        QueryFilter flowIdFilter = QueryFilter.builder()
+            .field(Field.FLOW_ID)
+            .operation(Op.EQUALS)
+            .value("my_flow")
+            .build();
+
+        // When / Then — the constraint only applies to EXECUTION
+        assertDoesNotThrow(() -> QueryFilter.validateQueryFilters(List.of(flowIdFilter), Resource.LOG));
+    }
```
[the remark this answers is not in the document]
Comment (rishi-jat) [reply core/src/test/java/io/kestra/core/models/QueryFilterTest.java:907]: same answer applies here ^^

File: core/src/main/java/io/kestra/core/models/QueryFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/models/QueryFilter.java:888]: We can leave with it.

File: core/src/test/java/io/kestra/core/models/QueryFilterTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (loicmathieu) [reply core/src/test/java/io/kestra/core/models/QueryFilterTest.java:907]: Yeah

**Comment by loicmathieu** [comment]
@rishi-jat I updated the PR description to list this PR as part of the issue and not fixes it as only merging the index would fix it. Before I merge it, can you rebase it?

I would also need to do the same change on our EE Elasticsearch repository.

**Comment by rishi-jat** [comment]
> @rishi-jat I updated the PR description to list this PR as part of the issue and not fixes it as only merging the index would fix it. Before I merge it, can you rebase it?
> 
> I would also need to do the same change on our EE Elasticsearch repository.

Doing!

**Comment by rishi-jat** [comment]
CC @loicmathieu

**Review by loicmathieu — APPROVED** [review]
LGTM, thank

**Comment by loicmathieu** [comment]
@rishi-jat finally, it was decided not to pursue in that direction. We prefer not to limit what a user can filter on, even if under the cover we didn't have performant index for such filter.

Sorry about that and thanks a lot for your contribution.
